docs/blueprints/EP033.rst

```diff
@@ -0,0 +1,252 @@
+:EP: 33
+:Title: A strategy for creating proxy ports to support telemetry-enabled Ethernet Virtual Circuits
+:Authors:
+    - Jeronimo Bezerra <jbezerra AT fiu DOT edu>
+    - Italo Valcy <idasilva AT fiu DOT edu>
+    - Vinicius Arcanjo <vindasil AT fiu DOT edu>
+:Created: 2022-09-03
+:Kytos-Version: 2022.3
+:Status: Draft
+
+
+*******************************************************
+EP033 - Creating proxy ports for telemetry-enabled EVCs
+*******************************************************
+
+
+Abstract
+========
+
+EP033 describes the creation and maintenance of **proxy ports** to support EP031 when creating telemetry-enabled EVCs.
+
+Motivation
+==========
+
+EP031's item I.4 states that EP31 won't address how **proxy ports** will be handled. However, **proxy ports** are a main requirement for an INT deployment when using NoviWare's pipeline. EP033 aims to fill the gap and address how proxy ports should be created and maintained by Kytos. Blueprint EP33 suggests leveraging the Kytos-ng `topology` napp's resources to implement **proxy ports**.
+
+I. When to use proxy ports
+==========================
+
+When deploying INT-enabled EVCs for the NoviWare's pipeline, to be capable of adding metadata and sending reports at the same INT hop, a packet has to be re-injected into the pipeline via external loops.
+
+For this document, external loops are physical connections on which both sides of the connection are in the same network device or switch. To increase **scalability**, since **proxy ports** are only used in one direction and at the INT Sink switch, **proxy ports** can be created with a single QSFP/QSFP28 transceiver and a single optical fiber connecting TX (transmission) to RX (reception). This kind of connection is also known as a loopback connection.
+
+There are three ways of handling the need for the **proxy port** concept:
+
+1) Not using **proxy ports** at all.
+
+In networks where most or all traffic is transported over inter-switch EVCs (EVCs where UNIs are in different switches), **proxy ports** can be ignored since the `add_int_metadata` action will be executed by the penultimate INT hop only. This is acceptable because all NNI ports would be monitored with INT. However, if intra-EVCs exist in the INT Sink switch, network visibility is compromised without **proxy ports** since intra-EVCs could be responsible for bottlenecks and these bottlenecks wouldn't be detected. This approach can be used when loopback connections are not available on the INT Sink Switch.
```

Comment:
Can you please provide more details on the statement "acceptable because all NNI ports would be monitored with INT" ? Is this because each traffic direction would help monitor the missing NNI?

Reply:
There is no missing NNI. Since all EVCs are inter-switch (meaning, links/NNI are used) and all links/NNI are monitored by INT, we can infer the final destination and sum the traffic going to a specific UNI in the destination switch.

Reply:
Please consider the following EVC between A and B and the path as described (Xi are the switches along the path, numbers inside parentheses are the port numbers):
```
A --> (1) X1 (2) --> (2) X2 (3) --> (3) X3 (4) --> (4) X4 (5) --> B
B --> (5) X4 (4) --> (4) X3 (3) --> (3) X2 (2) --> (2) X1 (1) --> A
```

If we don't have proxy ports in the above scenarios, how will we have the INT monitoring on the X4 NNI ingress interface (X4:4) and X1 NNI ingress interface (X1:2)?

Reply:
X3 will use add_int_metadata on port 4 which will monitor the incoming traffic on X4:4.
X4 will use add_int_metadata on port 4 which will monitor the incoming traffic on X3:4.
X1 will use add_int_metadata on port 2 which will monitor the incoming traffic on X2:2.
X2 will use add_int_metadata on port 2 which will monitor the incoming traffic on X1:2. The proxy would would be used in case you decide to monitor port X4:5 (B) or X1:1 (A). But since all traffic going to X4:5  (B) come from port X4:4, you know what's coming.

Reply:
Thank you for all the details and also for the discussion we have about this! It was clear to me that the goal is not make the proxy port a mandatory requirement for the INT napp (except for the intra-switch EVC, which wouldn't work without it). In the case of the inter-switch EVC, the last hop NNI information can be easily inferred from the previous hop depending on the link technology. So, even though it is very useful to have the proxy port, it is not mandatory.